specs/_features/maxeb_increase/capella.py

```diff
@@ -1776,6 +1790,49 @@ def process_execution_layer_withdraw_request(
     ))
 
 
+def is_consolidated(validator: Validator) -> bool:
+   validator.consolidated_to != UNSET_CONSOLIDATED_TO
+
+
+def process_consolidation(state: BeaconState, consolidation: Consolidation) -> None:
+    target_validator = state[consolidation.target_index]
+    source_validator = state[consolidation.source_index]
+
+    assert is_active_validator(target_validator) and not target_validator.slashed and not is_consolidated(target_validator)
+    assert not source_validator.slashed and not is_consolidated(source_validator)
+
+    # verify source withdrawal credentials, which have authority over validating keys
+    assert source_validator.withdrawal_credentials[:1] == ETH1_ADDRESS_WITHDRAWAL_PREFIX
+    assert source_validator.withdrawal_credentials[12:] == consolidation.source_address
```

Comment:
A mechanism similar to EIP-7002 should add the consolidation message on-chain, setting msg.sender to source address. A consolidation operation is only available to validators with 0x01 credentials.